### ISSUE TITLE
Investigate behavior of packages when rendering Qmd documents with and without renv

### ISSUE THREAD

**Issue by torwhite** [UCSBCarpentry/Reproducible-Publications-with-RStudio-Quarto#21]
Need to spend some more time testing out the renv virtual environments. 

Also, rendering Quarto documents without virtual environment AND without installing required packages seems to work? Need to look into this more as well.

**Comment by rcurty** [comment]
https://rstudio.github.io/renv/articles/renv.html

**Comment by rcurty** [comment]
Greg via Slack:
So, renv is pretty nifty.  I simulated a platform change by un-installing a package I previously had been using.  When I restarted my project, renv noticed right away:
* Project '~/Desktop/workshop' loaded. [renv 0.16.0]
[Workspace loaded from ~/Desktop/workshop/.RData]

* The project may be out of sync -- use `renv::status()` for more details.
And then:
renv::status()
The following package(s) are recorded in the lockfile, but not installed:
                 _
  palmerpenguins   [0.1.1]

Use `renv::restore()` to install these packages.
A call to renv::restore reinstalled palmerpenguins.  Easy-peasy.  I'll add this to the lesson at an appropriate place.

One hitch with renv is that I have yet to get it to update my .gitignore, which is fairly important.  Still trying.

I can't recommend including renv, at least not this go-around.  It's more a developer tool that can be used with some care and fiddling, and is not bullet-proof and is too complicated for beginners.  The fact that it doesn't modify .gitignore as claimed is problematic.  The renv/library folder contains host-specific pathnames.  If it gets included in the git repo, the effect is that people will wipe out each other's R projects.  Things can be patched up by running renv::init again, but that is counter to the reason one is using renv in the first place.
